cinema/views.py

```diff
@@ -21,118 +22,161 @@
+        serializer = CinemaHallSerializer(cinema_halls, many=True)
+        return Response(serializer.data)
+
+    def create(self, request):
+        serializer = CinemaHallSerializer(data=request.data)
+        if serializer.is_valid(raise_exception=True):
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class MovieViewSet(viewsets.ViewSet):
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
+
+    def list(self, request):
+        movies = Movie.objects.prefetch_related("genres", "actors")
+        serializer = MovieListSerializer(movies, many=True)
+        return Response(serializer.data)
+
+    def retrieve(self, request, pk=None):
+        movie = Movie.objects.prefetch_related("genres", "actors").get(pk=pk)
+        serializer = MovieDetailSerializer(movie)
+        return Response(serializer.data)
+
+    def create(self, request):
+        serializer = MovieSerializer(data=request.data)
+        if serializer.is_valid(raise_exception=True):
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class MovieSessionViewSet(viewsets.ViewSet):
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
+
+    def list(self, request):
+        movie_sessions = MovieSession.objects.all()
+        serializer = MovieSessionListSerializer(movie_sessions, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def retrieve(self, request, pk=None):
+        try:
+            movie_session = MovieSession.objects.get(pk=pk)
+            serializer = MovieSessionDetailSerializer(movie_session)
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        except MovieSession.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+    def create(self, request):
+        serializer = MovieSessionSerializer(data=request.data)
+        if serializer.is_valid(raise_exception=True):
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def update(self, request, pk=None):
+        try:
+            movie_session = MovieSession.objects.get(pk=pk)
+            serializer = MovieSessionSerializer(
+                movie_session, data=request.data
+            )
+            if serializer.is_valid(raise_exception=True):
+                serializer.save()
+                return Response(serializer.data, status=status.HTTP_200_OK)
+            return Response(
+                serializer.errors, status=status.HTTP_400_BAD_REQUEST
+            )
```

Comment:
it you use `is_valid` with raise_exception=True you shouldn't use if. Fix everywhere, please
```suggestion
            serializer.is_valid(raise_exception=True):
            serializer.save()
            return Response(serializer.data, status=status.HTTP_200_OK)
```